### PR TITLE
E2E Tests: Updates the wait function name to discourage its usage

### PR DESCRIPTION
**Pull request by youknowriad**
related https://github.com/WordPress/gutenberg/pull/6822#pullrequestreview-122815920

### CODE REVIEW

File: test/e2e/support/utils.js [hunk 1]
```diff
@@ -107,6 +107,6 @@ export async function pressWithModifier( modifier, key ) {
  *
  * @return {Promise} Promise resolving after the given timeout
  */
-export async function wait( timeout ) {
+export async function waitDurationYesImCertainWaitForSelectorIsNotABetterOption( timeout ) {
```
Comment (youknowriad) [inline test/e2e/support/utils.js]: I wonder if we could just name it `waitForAnimation` (that's the current use-case) Or maybe just leave it as `wait` and add a comment.

**Comment by noisysocks** [comment]
Why do define our own `wait` function? Puppeteer has [`page.waitFor()`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitforselectororfunctionortimeout-options-args) which can accept a timeout number.

(But yeah, we should discourage its usage.)

**Comment by youknowriad** [comment]
@noisysocks Actually, I missed this one

**Comment by aduth** [comment]
Could we set ESLint to at least warn of its use?

**Review by aduth — APPROVED** [review]
Nice 👍 Thanks

**Comment by aduth** [comment]
Discovering now that the [predicate argument form](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitforselectororfunctionortimeout-options-args) of this function can be useful in some cases, e.g. waiting for some data to resolve via `wp.data.select( 'core/data' ).hasFinishedResolution`.

Improvements suggested as part of: #13706

**Pull request merged**